### DISCUSSION TITLE
Possibility to deactivate marketplace

### DISCUSSION THREAD

**Discussion by the-other-dev** [directus/directus#24879] [Feature Requests]
### Describe the Improvement

Is it possible to add a configuration option to deactivate the marketplace?

Deactivating includes the following:
* Marketplace menu item is not displayed in the settings menu

Background

In self-hosted situations you might not be able to use the (public) marketplace due to compliance reasons.

Workarounds

a) Configure your own marketplace and implement the API to serve an empty marketplace.
b) Deny access (e.g. via network policies) to the public marketplace and receive a timeout error/internal error in the UI when clicking on the marketplace menu item (slightly bad user experience)

**Comment by robinsonjohn** [comment]
I agree. This is similar to the ability for WordPress to allow plugin installations via the UI. It allows software to be installed on-the-fly in a production environment, which is a huge "no-no".

Utilizing self-hosted Directus, I prefer to manually add the extension's Git URL to `package.json` `devDependencies`, and use a script to copy the files to the `/extensions` directory. This allows everything to be tested locally and keeps all app environments in-sync.

While the marketplace has its benefits for some users, it makes it impossible to keep all app environments (i.e. dev, staging, etc) in-sync and can introduce problems directly into the production environment.

**Comment by achille2k** [comment]
I agree

**Comment by YannickArmspach** [comment]
Agreed. In the meantime, you can use the variable `EXTENSIONS_LIMIT=0` to disable the installation ability.
